pinot-query-planner/src/main/java/org/apache/pinot/query/routing/WorkerMetadata.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.routing;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.routing.TimeBoundaryInfo;
+
+
+/**
+ * The {@code WorkerMetadata} info contains the information for executing a particular worker on a particular stage.
+ *
+ * <p>It contains information about:</p>
+ * <ul>
+ *   <li>the tables it is suppose to scan (including real-time and offline portions separately)</li>
+ *   <li>the underlying segments a stage requires to execute upon if it is a leaf stage worker</li>
+ *   <li>the partition mechanism of the data being execute on this worker</li>
+ * </ul>
+ *
+ * <p>WorkerMetadata should allow {@link org.apache.pinot.query.planner.stage.MailboxSendNode} and
+ * {@link org.apache.pinot.query.planner.stage.MailboxReceiveNode} to derive the appropriate set of mailboxes to be
+ * created when executing the worker.</p>
+ */
+public class WorkerMetadata {
+  private final QueryServerAddress _queryServerAddress;
+  private final TableInfo _tableInfo;
```

Comment:
additional fields will be put in later for WorkerMetadata such as `PartitionInfo`
we can potentially add more info for optimizations on server runtime